engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceMigrationPreconditionChecker.java

```diff
@@ -357,6 +361,50 @@ public static void requireSameElementType(
     }
   }
 
+  /**
+   * Since we introduce zeebe user tasks and job worker tasks has the same bpmn element type, we
+   * need to check whether the given element instance and target element has the same user task
+   * type. Throws an exception if they have different types.
+   *
+   * @param targetProcessDefinition target process definition to retrieve the target element type
+   * @param targetElementId target element id
+   * @param elementInstance element instance to do the check
+   * @param processInstanceKey process instance key to be logged
+   */
+  public static void requireSameUserTaskType(
+      final DeployedProcess targetProcessDefinition,
+      final String targetElementId,
+      final ElementInstance elementInstance,
+      final long processInstanceKey) {
+    final ProcessInstanceRecord elementInstanceRecord = elementInstance.getValue();
+    final AbstractFlowElement targetElement =
+        targetProcessDefinition.getProcess().getElementById(targetElementId);
+    final BpmnElementType targetElementType = targetElement.getElementType();
+
+    if (elementInstanceRecord.getBpmnElementType() == BpmnElementType.USER_TASK
+        && targetElementType == BpmnElementType.USER_TASK) {
+      final ExecutableUserTask targetUserTask =
+          (ExecutableUserTask) targetProcessDefinition.getProcess().getElementById(targetElementId);
```

Comment:
🔧 Rather than a direct cast you can use `getElementById(targetElementId, ExecutableUserTask.class)`